backend-rust/src/indexer.rs

```diff
@@ -1295,6 +1304,169 @@ impl PreparedAccountCreation {
+            },
+            DelegationEvent::DelegationRemoved {
+                delegator_id,
+            } => PreparedAccountDelegationEvent::Removed {
+                account_id: delegator_id.id.index.try_into()?,
+            },
+            DelegationEvent::BakerRemoved {
+                baker_id,
+            } => PreparedAccountDelegationEvent::BakerRemoved {
+                baker_id: baker_id.id.index.try_into()?,
+            },
+        };
+        Ok(prepared)
+    }
+
+    async fn save(
+        &self,
+        tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
+    ) -> anyhow::Result<()> {
+        match self {
+            PreparedAccountDelegationEvent::StakeIncrease {
+                account_id,
+                staked,
+            }
+            | PreparedAccountDelegationEvent::StakeDecrease {
+                account_id,
+                staked,
+            } => {
+                sqlx::query!(
+                    r#"UPDATE accounts SET delegated_stake = $1 WHERE index = $2"#,
+                    staked,
+                    account_id
+                )
+                .execute(tx.as_mut())
+                .await?;
+            }
+            PreparedAccountDelegationEvent::Added {
+                account_id,
+            }
+            | PreparedAccountDelegationEvent::Removed {
+                account_id,
+            } => {
+                sqlx::query!(
+                    r#"UPDATE accounts SET delegated_stake = 0, delegated_restake_earnings = false, delegated_target_baker_id = NULL WHERE index = $1"#,
+                    account_id
+                )
+                .execute(tx.as_mut())
+                .await?;
+            }
+
+            PreparedAccountDelegationEvent::SetRestakeEarnings {
+                account_id,
+                restake_earnings,
+            } => {
+                sqlx::query!(
+                    r#"UPDATE accounts SET delegated_restake_earnings = $1 WHERE index = $2"#,
+                    *restake_earnings,
+                    account_id
+                )
+                .execute(tx.as_mut())
+                .await?;
+            }
+            PreparedAccountDelegationEvent::SetDelegationTarget {
+                account_id,
+                target_id,
+            } => {
+                sqlx::query!(
+                    r#"UPDATE accounts SET delegated_target_baker_id = $1 WHERE index = $2"#,
+                    *target_id,
+                    account_id
+                )
+                .execute(tx.as_mut())
+                .await?;
+            }
+            PreparedAccountDelegationEvent::BakerRemoved {
+                baker_id,
+            } => {
+                sqlx::query!(
+                    r#"UPDATE accounts SET delegated_stake = 0, delegated_restake_earnings = false, delegated_target_baker_id = NULL WHERE index = $1"#,
+                    *baker_id
+                )
+                .execute(tx.as_mut())
+                .await?;
```

Comment:
I don't this update is needed, since the _delegation event_ of baker removed would be followed by events updating these

Reply:
I would like to keep it as is. This is the what CCDScan already does, and I would like to be the difference of behaviour between the two services as small as possible. Then we can factor it out later if need be.

Since this is only happening on the indexer, the change is speed would not impact the behaviour at all.

Reply:
I assume a baker cannot `delegate its stake` somewhere else since it is baking itself. If true, then the fields `delegated_...` should already be not set for a valid baker when the `BakerRemoved` event is logged.

Reply:
https://github.com/Concordium/concordium-scan/blob/516a1c3256555f8a4cedc6f50bd8db231fe4276e/backend/Application/Api/GraphQL/Import/DelegationImportHandler.cs#L196

As mentioned this is what we do, I prefer to align as much as possible with existing behaviour. 

But yes it sounds as if we are just making a baker removed twice.